src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -22,6 +24,35 @@ def daemon_check_post(self, daemon_descrs: List[DaemonDescription]):
         """The post actions needed to be done after daemons are checked"""
         raise NotImplementedError()
 
+    def _get_addr(self, hostname: str):
+        """Get a host's address with its hostname."""
+        return self.mgr.inventory.get_addr(hostname)
+
+    def _set_service_url_on_dashboard(self,
+                                      service_name: str,
+                                      get_mon_cmd: str,
+                                      set_mon_cmd: str,
+                                      service_url: str):
+        """A helper to get and set service_url via Dashboard's MON command."""
+        try:
+            _, out, _ = self.mgr.check_mon_command({
+                'prefix': get_mon_cmd
+            })
+        except MonCommandFailed as e:
+            logger.warning('Failed to get service URL for %s: %s', service_name, e)
+            return
```

Comment:
that's interesting. We also have `mgr.mon_command` that doesn't raise and exception. Haven't expected to see `check_mon_command` used this way. I like it, but could you please add `r.retval` to the message of `MonCommandFailed`?

Reply:
Add in a separate commit: https://github.com/ceph/ceph/pull/35194/commits/6e4d4374aedc725975a8fcbb458e88bfb32c099a